src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -451,6 +513,17 @@ protected void fillRhsSensitivityVariable(LfNetwork lfNetwork, EquationSystem eq
         }
     }
 
+    private void setPredefinedResults(Collection<LfSensitivityFactor> lfFactors, GraphDecrementalConnectivity<LfBus> connectivity, int mainComponent) {
+        for (LfSensitivityFactor factor : lfFactors) {
+            // check if the factor function and variable are in different connected components
+            if (factor.areVariableAndFunctionDisconnected(connectivity)) {
+                factor.setPredefinedResult(0d);
+            } else if (!factor.isConnectedToComponent(mainComponent, connectivity)) {
```

Comment:
You always call `isConnectedToComponent` with mainComponent, so in fact you could rename it to `isConnectedToMaintComponent` and use `getSmallComponents()` instead of `getComponentNumber()`

Reply:
See previous conversation